tripal_chado/src/Task/ChadoPreparer.php

```diff
@@ -158,6 +191,624 @@ public function performTask() :bool {
+      ],
+    ];
+    
+    chado_create_custom_table('tripal_gff_temp', $schema, TRUE, NULL, 
+      FALSE, $this->chado);
+  }
+  
+  /**
+   *
+   */
+  function tripal_chado_add_tripal_gffcds_temp_table() {
+    $schema = [
+      'table' => 'tripal_gffcds_temp',
+      'fields' => [
+        'feature_id' => [
+          'type' => 'int',
+          'not null' => TRUE,
+        ],
+        'parent_id' => [
+          'type' => 'int',
+          'not null' => TRUE,
+        ],
+        'phase' => [
+          'type' => 'int',
+          'not null' => FALSE,
+        ],
+        'strand' => [
+          'type' => 'int',
+          'not null' => TRUE,
+        ],
+        'fmin' => [
+          'type' => 'int',
+          'not null' => TRUE,
+        ],
+        'fmax' => [
+          'type' => 'int',
+          'not null' => TRUE,
+        ],
+      ],
+      'indexes' => [
+        'tripal_gff_temp_idx0' => ['feature_id'],
+        'tripal_gff_temp_idx0' => ['parent_id'],
+      ],
+    ];
+    chado_create_custom_table('tripal_gffcds_temp', $schema, TRUE, NULL, 
+      FALSE, $this->chado);
+  }
+  
+  /**
+   *
+   */
+  function tripal_chado_add_tripal_gffprotein_temp_table() {
+    $schema = [
+      'table' => 'tripal_gffprotein_temp',
+      'fields' => [
+        'feature_id' => [
+          'type' => 'int',
+          'not null' => TRUE,
+        ],
+        'parent_id' => [
+          'type' => 'int',
+          'not null' => TRUE,
+        ],
+        'fmin' => [
+          'type' => 'int',
+          'not null' => TRUE,
+        ],
+        'fmax' => [
+          'type' => 'int',
+          'not null' => TRUE,
+        ],
+      ],
+      'indexes' => [
+        'tripal_gff_temp_idx0' => ['feature_id'],
+        'tripal_gff_temp_idx0' => ['parent_id'],
+      ],
+      'unique keys' => [
+        'tripal_gff_temp_uq0' => ['feature_id'],
+      ],
+    ];
+    chado_create_custom_table('tripal_gffprotein_temp', $schema, TRUE, NULL, 
+      FALSE, $this->chado);
+  }
+  
+  /**
+   * Creates a temporary table to store obo details while loading an obo file
+   *
+   */
+  function tripal_chado_add_tripal_obo_temp_table() {
+    // the tripal_obo_temp table is used for temporary housing of records when loading OBO files
+    // we create it here using plain SQL because we want it to be in the chado schema but we
+    // do not want to use the Tripal Custom Table API because we don't want it to appear in the
+    // list of custom tables.  It needs to be available for the Tripal Chado API so we create it
+    // here and then define it in the tripal_cv/api/tripal_cv.schema.api.inc
+    if (!chado_table_exists('tripal_obo_temp')) {
+      $sql = "
+        CREATE TABLE {tripal_obo_temp} (
+          id character varying(255) NOT NULL,
+          stanza text NOT NULL,
+          type character varying(50) NOT NULL,
+          CONSTRAINT tripal_obo_temp_uq0 UNIQUE (id)
+        );
+      ";
+      chado_query($sql);
```

Comment:
We should use the new `$this->chado->query()` here I believe... And for the index creation next too.

Reply:
Yes we should!  I'll fix it.